src/internal_telemetry/allocations/mod.rs

```diff
@@ -36,10 +43,11 @@ static GROUP_MEM_STATS: [[CachePadded<AtomicI64>; NUM_GROUPS]; 8] = [
     arr![CachePadded::new(AtomicI64::new(0)); 128],
 ];
 
+// TODO: Replace this with [`std::thread::ThreadId::as_u64`] once it is stabilized.
 static THREAD_COUNTER: AtomicUsize = AtomicUsize::new(0);
 
 thread_local! {
-    static THREAD_ID: AtomicUsize = AtomicUsize::new(THREAD_COUNTER.fetch_add(1, Ordering::SeqCst) + 1);
+    static THREAD_ID: usize = THREAD_COUNTER.fetch_add(1, Ordering::Relaxed) + 1;
```

Comment:
I don't think we want the `+ 1` here.

This will lead to us not using the 0th slot in `GROUP_MEM_STATS` until we exceed 8 threads that go through this codepath.

Reply:
Ah agreed